core/src/main/scala/com/gu/tableversions/core/model.scala

```diff
@@ -12,16 +12,15 @@ import cats.effect.IO
   */
 final case class Partition(columnValues: List[Partition.ColumnValue]) {
 
+  require(columnValues.nonEmpty, "Must have at least one partition column")
+
```

Comment:
Can we use `NonEmptyList` rather than relying on runtime check?

Reply:
Yes we talked about that, we think there are a number of places we can do this but we decided it would be a step to far for this refactoring.

Reply:
@TBonnin On reflection, it seemed better to include this refactoring here so we did as you suggested.